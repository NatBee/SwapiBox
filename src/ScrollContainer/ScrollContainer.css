.scroll-container {
  background-color: green; }
