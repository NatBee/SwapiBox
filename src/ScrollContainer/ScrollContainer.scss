.scroll-container {
  background-color: #8E7842;
  height: 50px;
  padding: 20px;
  color: white;
  display: flex;
  justify-content: center;
  perspective: 400px;
  color: #F7F4DB;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
}

.fade {
  position: relative;
  width: 100%;
  min-height: 75vh;
  background-image: linear-gradient(0deg, transparent, black 90%);
  z-index: 1;
  overflow: hidden;
}

.crawl {
  position: relative;
  top: 200px;
  transform-origin: 50% 100%;
  animation: crawl 30s linear;
}

@keyframes crawl {
  0% {
    top: -1;
    transform: rotateX(15deg) translateZ(0);
  }
  100% { 
    top: -5000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}