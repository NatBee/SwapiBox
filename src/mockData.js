const mockData = {
  "peopleData": [
    {
    "name": "Beru Whitesun lars",
    "homeworld": undefined,
    "population": undefined,
    "species": undefined,
    "favorite": false
    }
  ],
  "planetData": [
    {
      "name": "Geonosis",
      "climate": "temperate, arid",
      "population": "100000000000",
      "terrain": "rock, desert, mountain, barren",
      "residents": [
        undefined
      ],
      "favorite": false
    }
  ],
  "vehicleData": [ 
    {
      "name": "Sail barge",
      "model": "Modified Luxury Sail Barge",
      "class": undefined,
      "passengers": "500"
    }
   ],
   "movieData": [ 
   {
      "title": "Revenge of the Sith",
      "releaseDate": "2005-05-19",
      "openingCrawl": "War! The Republic is crumbling\r\nunder attacks by the ruthless\r\nSith Lord, Count Dooku.\r\nThere are heroes on both sides.\r\nEvil is everywhere.\r\n\r\nIn a stunning move, the\r\nfiendish droid leader, General\r\nGrievous, has swept into the\r\nRepublic capital and kidnapped\r\nChancellor Palpatine, leader of\r\nthe Galactic Senate.\r\n\r\nAs the Separatist Droid Army\r\nattempts to flee the besieged\r\ncapital with their valuable\r\nhostage, two Jedi Knights lead a\r\ndesperate mission to rescue the\r\ncaptive Chancellor...."
    }
  ]

}

export default mockData;