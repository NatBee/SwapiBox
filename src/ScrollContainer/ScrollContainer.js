import React from 'react';
import logo from '../images/Star-Wars-Logo.png';
import './ScrollContainer.css';

const ScrollContainer = ({handleClickFavorites, favorites, movieData}) => {
  const numberOfFavorites = favorites.length;
  const openingScroll = movieData.map( data => {
    return (
      <section>
        <p className="opening-crawl">{data.openingCrawl}</p>
        <p className="title">{data.title}</p>
        <p className="release-date">{data.releaseDate}</p>
      </section>
    )
  }); 

  return (
    <section>
      <section className="scroll-container">
        {openingScroll}
      </section>
      <img  src={logo} 
            className="App-logo" 
            alt="logo" 
      />
      <h1 className="App-title">
        Lego Star Wars
      </h1>
      <button className="favorites" onClick={ () => handleClickFavorites()}>
        View Favorites 
        <div>
          {numberOfFavorites}
        </div>
      </button>
    </section>
  )
} 

export default ScrollContainer;