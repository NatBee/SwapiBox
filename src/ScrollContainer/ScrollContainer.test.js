/* eslint-disable */ 
import React from 'react';
import ReactDOM from 'react-dom';
import ScrollContainer from './ScrollContainer.js';
import { shallow } from 'enzyme';
import mockData from '../mockData.js'

describe('ScrollContainer', () => {
  let mockFavorites;
  let mockMovie;
  let renderedComponent;

  beforeAll(() => {
    mockFavorites = mockData.favorites;
    mockMovie = mockData.movieData;
    renderedComponent = shallow(<ScrollContainer 
                                  favorites={mockFavorites} 
                                  movieData={mockMovie} 
                                />)
  })

  it('should match the snapshot', () => {
    expect(renderedComponent).toMatchSnapshot()
  })

  it('should have 1 .opening-crawl tag', () => {
    expect(renderedComponent.find(".opening-crawl").length).toEqual(1);
  })

})