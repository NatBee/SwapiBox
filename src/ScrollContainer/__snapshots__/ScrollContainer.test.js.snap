// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollContainer
    favorites={
        Array [
            Object {
              "climate": "temperate, arid",
              "favorite": true,
              "name": "Geonosis",
              "population": "100000000000",
              "residents": Array [
                undefined,
              ],
              "terrain": "rock, desert, mountain, barren",
            },
          ]
    }
    movieData={
        Array [
            Object {
              "openingCrawl": "War! The Republic is crumbling
        under attacks by the ruthless
        Sith Lord, Count Dooku.
        There are heroes on both sides.
        Evil is everywhere.
        
        In a stunning move, the
        fiendish droid leader, General
        Grievous, has swept into the
        Republic capital and kidnapped
        Chancellor Palpatine, leader of
        the Galactic Senate.
        
        As the Separatist Droid Army
        attempts to flee the besieged
        capital with their valuable
        hostage, two Jedi Knights lead a
        desperate mission to rescue the
        captive Chancellor....",
              "releaseDate": "2005-05-19",
              "title": "Revenge of the Sith",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          class="fade"
/>,
        <section
          className="scroll-container"
>
          <section
                    className="crawl"
          >
                    <p
                              className="opening-crawl"
                    >
                              War! The Republic is crumbling
                              under attacks by the ruthless
                              Sith Lord, Count Dooku.
                              There are heroes on both sides.
                              Evil is everywhere.
                              
                              In a stunning move, the
                              fiendish droid leader, General
                              Grievous, has swept into the
                              Republic capital and kidnapped
                              Chancellor Palpatine, leader of
                              the Galactic Senate.
                              
                              As the Separatist Droid Army
                              attempts to flee the besieged
                              capital with their valuable
                              hostage, two Jedi Knights lead a
                              desperate mission to rescue the
                              captive Chancellor....
                    </p>
                    <p
                              className="title"
                    >
                              Revenge of the Sith
                    </p>
                    <p
                              className="release-date"
                    >
                              2005-05-19
                    </p>
          </section>
</section>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "class": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section
              className="crawl"
>
              <p
                            className="opening-crawl"
              >
                            War! The Republic is crumbling
                            under attacks by the ruthless
                            Sith Lord, Count Dooku.
                            There are heroes on both sides.
                            Evil is everywhere.
                            
                            In a stunning move, the
                            fiendish droid leader, General
                            Grievous, has swept into the
                            Republic capital and kidnapped
                            Chancellor Palpatine, leader of
                            the Galactic Senate.
                            
                            As the Separatist Droid Army
                            attempts to flee the besieged
                            capital with their valuable
                            hostage, two Jedi Knights lead a
                            desperate mission to rescue the
                            captive Chancellor....
              </p>
              <p
                            className="title"
              >
                            Revenge of the Sith
              </p>
              <p
                            className="release-date"
              >
                            2005-05-19
              </p>
</section>,
          ],
          "className": "scroll-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="opening-crawl"
>
                  War! The Republic is crumbling
                  under attacks by the ruthless
                  Sith Lord, Count Dooku.
                  There are heroes on both sides.
                  Evil is everywhere.
                  
                  In a stunning move, the
                  fiendish droid leader, General
                  Grievous, has swept into the
                  Republic capital and kidnapped
                  Chancellor Palpatine, leader of
                  the Galactic Senate.
                  
                  As the Separatist Droid Army
                  attempts to flee the besieged
                  capital with their valuable
                  hostage, two Jedi Knights lead a
                  desperate mission to rescue the
                  captive Chancellor....
</p>,
                <p
                  className="title"
>
                  Revenge of the Sith
</p>,
                <p
                  className="release-date"
>
                  2005-05-19
</p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "War! The Republic is crumbling
under attacks by the ruthless
Sith Lord, Count Dooku.
There are heroes on both sides.
Evil is everywhere.

In a stunning move, the
fiendish droid leader, General
Grievous, has swept into the
Republic capital and kidnapped
Chancellor Palpatine, leader of
the Galactic Senate.

As the Separatist Droid Army
attempts to flee the besieged
capital with their valuable
hostage, two Jedi Knights lead a
desperate mission to rescue the
captive Chancellor....",
                  "className": "opening-crawl",
                },
                "ref": null,
                "rendered": "War! The Republic is crumbling
under attacks by the ruthless
Sith Lord, Count Dooku.
There are heroes on both sides.
Evil is everywhere.

In a stunning move, the
fiendish droid leader, General
Grievous, has swept into the
Republic capital and kidnapped
Chancellor Palpatine, leader of
the Galactic Senate.

As the Separatist Droid Army
attempts to flee the besieged
capital with their valuable
hostage, two Jedi Knights lead a
desperate mission to rescue the
captive Chancellor....",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Revenge of the Sith",
                  "className": "title",
                },
                "ref": null,
                "rendered": "Revenge of the Sith",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2005-05-19",
                  "className": "release-date",
                },
                "ref": null,
                "rendered": "2005-05-19",
                "type": "p",
              },
            ],
            "type": "section",
          },
        ],
        "type": "section",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            class="fade"
/>,
          <section
            className="scroll-container"
>
            <section
                        className="crawl"
            >
                        <p
                                    className="opening-crawl"
                        >
                                    War! The Republic is crumbling
                                    under attacks by the ruthless
                                    Sith Lord, Count Dooku.
                                    There are heroes on both sides.
                                    Evil is everywhere.
                                    
                                    In a stunning move, the
                                    fiendish droid leader, General
                                    Grievous, has swept into the
                                    Republic capital and kidnapped
                                    Chancellor Palpatine, leader of
                                    the Galactic Senate.
                                    
                                    As the Separatist Droid Army
                                    attempts to flee the besieged
                                    capital with their valuable
                                    hostage, two Jedi Knights lead a
                                    desperate mission to rescue the
                                    captive Chancellor....
                        </p>
                        <p
                                    className="title"
                        >
                                    Revenge of the Sith
                        </p>
                        <p
                                    className="release-date"
                        >
                                    2005-05-19
                        </p>
            </section>
</section>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "class": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <section
                className="crawl"
>
                <p
                                className="opening-crawl"
                >
                                War! The Republic is crumbling
                                under attacks by the ruthless
                                Sith Lord, Count Dooku.
                                There are heroes on both sides.
                                Evil is everywhere.
                                
                                In a stunning move, the
                                fiendish droid leader, General
                                Grievous, has swept into the
                                Republic capital and kidnapped
                                Chancellor Palpatine, leader of
                                the Galactic Senate.
                                
                                As the Separatist Droid Army
                                attempts to flee the besieged
                                capital with their valuable
                                hostage, two Jedi Knights lead a
                                desperate mission to rescue the
                                captive Chancellor....
                </p>
                <p
                                className="title"
                >
                                Revenge of the Sith
                </p>
                <p
                                className="release-date"
                >
                                2005-05-19
                </p>
</section>,
            ],
            "className": "scroll-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="opening-crawl"
>
                    War! The Republic is crumbling
                    under attacks by the ruthless
                    Sith Lord, Count Dooku.
                    There are heroes on both sides.
                    Evil is everywhere.
                    
                    In a stunning move, the
                    fiendish droid leader, General
                    Grievous, has swept into the
                    Republic capital and kidnapped
                    Chancellor Palpatine, leader of
                    the Galactic Senate.
                    
                    As the Separatist Droid Army
                    attempts to flee the besieged
                    capital with their valuable
                    hostage, two Jedi Knights lead a
                    desperate mission to rescue the
                    captive Chancellor....
</p>,
                  <p
                    className="title"
>
                    Revenge of the Sith
</p>,
                  <p
                    className="release-date"
>
                    2005-05-19
</p>,
                ],
                "className": "crawl",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "War! The Republic is crumbling
under attacks by the ruthless
Sith Lord, Count Dooku.
There are heroes on both sides.
Evil is everywhere.

In a stunning move, the
fiendish droid leader, General
Grievous, has swept into the
Republic capital and kidnapped
Chancellor Palpatine, leader of
the Galactic Senate.

As the Separatist Droid Army
attempts to flee the besieged
capital with their valuable
hostage, two Jedi Knights lead a
desperate mission to rescue the
captive Chancellor....",
                    "className": "opening-crawl",
                  },
                  "ref": null,
                  "rendered": "War! The Republic is crumbling
under attacks by the ruthless
Sith Lord, Count Dooku.
There are heroes on both sides.
Evil is everywhere.

In a stunning move, the
fiendish droid leader, General
Grievous, has swept into the
Republic capital and kidnapped
Chancellor Palpatine, leader of
the Galactic Senate.

As the Separatist Droid Army
attempts to flee the besieged
capital with their valuable
hostage, two Jedi Knights lead a
desperate mission to rescue the
captive Chancellor....",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Revenge of the Sith",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "Revenge of the Sith",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2005-05-19",
                    "className": "release-date",
                  },
                  "ref": null,
                  "rendered": "2005-05-19",
                  "type": "p",
                },
              ],
              "type": "section",
            },
          ],
          "type": "section",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
